### ISSUE TITLE
Unable to deploy this component

### ISSUE THREAD

**Issue by Mike-Cleaver-LB** [SalesforceLabs/Configurable-Self-Registration-for-Experience-Cloud#38]
Hey team, We have this in our community but can't deploy it up because of the account recordtype id field

`The personAccountRecordTypeId property of component d18ddb32-0ac6-4cb0-9608-7aaa331c54f6 references an object with the ID value XXXXX. Occasionally, when deployed to a destination org, ID values can become invalid—for example, if the referenced ID doesn’t exist in the destination org. If you encounter component issues in your destination org, verify that the ID values are correct.`

So one solution is to remove the recordtype field from the json (component attributes) for example

```
"componentAttributes": {
						"accessLevelMode": "System",
						"accountId": "",
						"createNotFound": true,
						"customQuery": "SELECT id FROM Account WHERE firstname = :firstName AND lastname = :lastName AND personemail = :email",
						"errorOnCreate": "There was a problem registering your details. Please contact us for assistance.",
						"multipleRecordsFoundError": "There was a problem locating your details, please check them and try again or contact us for further assistance.",
						"noRecordFoundError": "There was a problem locating your details, please check them and try again or contact us for further assistance.",
						"objectCreateType": "Person Account",
						"passwordMatchError": "Passwords do not match",
						"comment": "\"personAccountRecordTypeId\": \"XXXXX\"",
						"portalLoginError": "There was a problem logging you into the portal. Please contact us for further assistance.",
						"portalRegistrationError": "There was a problem during registration. Please contact us for further assistance.",
						"portalRegistrationRedirect": "/",
						"portalRegistrationUserExists": "User is already registered. Please login to continue.",
						"registerButtonSignUpMessage": "Sign Up",
						"registerButtonWaitingMessage": "Registering... Please Wait.",
						"sendEmailConfirmation": false,
						"usernameTakenMessage": "The specified username is already taken. Please choose a unique username."
					},
```
					
but this means, every deployment, I have to reset that value.
Or we don't deploy the community at all (removing this page, results in the error above, for whatever reason)

Has anyone thought up a smart way to sort this?

Or are there any changes that can be made to the component, say have it find record type by name?

**Comment by jloweSFDC** [comment]
Hi @Mike-Cleaver-LB - you have a couple of options here...

1) If possible, which I appreciate it isn't always, the easiest way is to refresh the sandbox environments so that they all align with production - that way all Ids will be the same removing this issue.

2) If the above isn't possible, assuming you are deploying via CI/CD automated pipelines and using SFDX, you can use the built in SFDX String Replacement functionality. See: https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_ws_string_replace.htm for full information.

In your sfdx-project.json file you'd identify the correct directory/file to look in via the "filename" parameter i.e. the register.json file, look for the Id stored in the repository using the "stringToReplace" param and then replace with the higher environment value by specifying a file location that holds the correct Id for the given environment you're deploying to. Example:

![2024-10-01_09-03-31](https://github.com/user-attachments/assets/7fddff71-2ffa-448d-ad64-dfe28262a6f7)

**Comment by Mike-Cleaver-LB** [comment]
Superb thanks

**Comment by jloweSFDC** [comment]
Not sure on this. It's appears to be a deployment specific issue rather than the component. Does your deployment user have access to the recordTypeId that you have specified?

**Comment by Mike-Cleaver-LB** [comment]
yeah it's my user effectively. But that's fine. It could be gearset seeing, what I read as a warning, and failing the deployment because of it. The message is kinda vague "if your destination org has issues" not "your destination org will have issues" so I think it's meant to be an SFDX warning not an error.

so I am just gonna leave this as a manual fix post deployment going forward, and have the field blank our everytime.
We aren't deploying that frequently that this will be a big problem. and at some point I expect we will simply stop deploying the digital experience, and allow changes in production (this is a greenfields project atm)

**Comment by jloweSFDC** [comment]
Hi @Mike-Cleaver-LB - I've fixed this in version 1.9 coming soon. Watch this space for an updated version. If you're deploying an unmanaged version rather than the managed from the AppExchange, the code is available on a new branch. See the commit history here if you want to see how I've done it: https://github.com/SalesforceLabs/Configurable-Self-Registration-for-Experience-Cloud/commit/bcaaac1353fc32a3a01f8aaf449f48669fa2dce3.

**Comment by Mike-Cleaver-LB** [comment]
amazing. I can confirm ignoring warnings did re-enable the deployment, great spot!
I'm grateful you're still going ahead with the enhancement. It'd be good to re-enable this setting if possible.
